src/videojs-contrib-hls.js

```diff
@@ -569,17 +566,21 @@ class HlsHandler extends Component {
  */
 const HlsSourceHandler = function(mode) {
   return {
-    canHandleSource(srcObj) {
+    canHandleSource(srcObj, options) {
+      let settings = videojs.mergeOptions(videojs.options, options);
+
       // this forces video.js to skip this tech/mode if its not the one we have been
       // overriden to use, by returing that we cannot handle the source.
-      if (videojs.options.hls &&
-          videojs.options.hls.mode &&
-          videojs.options.hls.mode !== mode) {
+      if (settings.hls &&
+          settings.hls.mode &&
+          settings.hls.mode !== mode) {
         return false;
       }
-      return HlsSourceHandler.canPlayType(srcObj.type);
+      return HlsSourceHandler.canPlayType(srcObj.type, settings);
```

Comment:
It seems that `settings` and `options` are used interchangeably throughout the code (for instance, here `settings` are passed as the `options` parameter). I think it may make more sense to contextualize more. Maybe something like `mergedOptions` after merged, or `playerOptions` for player specific ones.